Sources/TuistCore/Graph/Mappers/GraphMapper.swift

```diff
@@ -6,21 +6,21 @@ import XcodeGraph
 public protocol GraphMapping {
     /// Given a value graph, it maps it into another value graph.
     /// - Parameter graph: Graph to be mapped.
-    func map(graph: Graph) async throws -> (Graph, [SideEffectDescriptor])
+    func map(graph: Graph, environment: MapperEnvironment) async throws -> (Graph, [SideEffectDescriptor], MapperEnvironment)
```

Comment:
In wonder if long-term we should evolve the API to be:
```suggestion
    func map(environment: MapperEnvironment) async throws -> (MapperEnvironment)
```

And merge graph and side effects into the environment.

Reply:
I think it's better to stay explicit about the inputs and outputs for now